rtl/eclrtl/rtldynfield.cpp

```diff
@@ -730,6 +772,8 @@ class CRtlFieldTypeDeserializer : public CInterfaceOf<IRtlFieldTypeDeserializer>
             {
                 const char *fieldName;
                 type.read(fieldName);
+                if (fieldName[0] == '\0')
+                    fieldName = nullptr;
```

Comment:
This probably leaks.

Reply:
Actually it is ok, but the code is a bit dubious - surprising that fieldName isn't cloned.

Reply:
Most of the code around here tries to avoid allocations if it is confident about lifetimes...